src/pipelines/data_gen.py

```diff
@@ -148,6 +147,72 @@ def run_lda(lda_topics, doc, vocab):
     docvector = vectorizer.transform([doc])
     return lda_topics.transform(docvector)[0]
 
+def w2v(doc, corpus, w2v_model, w2v, feature):
+    '''
+     Calculates Word2Vec features for a document and corpus
+
+     Args:
+         doc (str): the text of the document (before any preprocessing)
+         corpus (list): the text of the corpus
+         w2v_model (gensim.Word2Vec): Trained Word2Vec model
+         w2v (dict): Dictionary of Word2Vec parameters
+         feature (list): List of features extracted from text
+
+     Returns:
+         feature: List of features extracted from text
+     '''
+
+    docw2v = run_w2v(w2v_model, doc, w2v)
+    corpusw2v = run_w2v(w2v_model, corpus, w2v)
+    vectors = [docw2v, corpusw2v]
+    feature = gen_feature(vectors, w2v, feature)
+    return feature
+
+def run_w2v(w2v_model, doc, w2v):
+    '''
+      Calculates Word2Vec vectors for a document
```

Comment:
Maybe include some note in the docstring that this uses the first-and-last-sentence technique?